contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.3.0--4.4.0.sql

```diff
@@ -10566,6 +10566,50 @@ CALL sys.babelfish_alter_default_privilege_on_schema();
 -- Drop this procedure after it gets executed once.
 DROP PROCEDURE sys.babelfish_alter_default_privilege_on_schema();
 
+-- To handle the typmod values for procedure and function
+UPDATE pg_proc p1
+SET probin = 
+    CASE 
+        WHEN p1.probin::jsonb->'typmod_array' = '[]'::jsonb THEN p1.probin
+        ELSE (
+            SELECT (
+                jsonb_set(
+                    p1.probin::jsonb,
+                    '{typmod_array}',
+                    to_jsonb(
+                        (
+                            SELECT jsonb_agg(
+                                CASE
+                                    WHEN p2.prokind = 'p' OR (p2.prokind = 'f' AND p2.proallargtypes IS NULL AND p2.proargtypes[typ_index-1] IS NOT NULL) THEN
+                                        CASE
+                                            WHEN typmod::text = '-8000' AND p2.proargtypes[typ_index-1]::regtype in ('sys.varchar', 'sys.nvarchar', 'sys.varbinary') THEN '-1'
+                                            WHEN typmod::text = '-1' AND p2.proargtypes[typ_index-1]::regtype::text = 'sys.smalldatetime' THEN '0'
+                                            WHEN typmod::text = '-1' AND p2.proargtypes[typ_index-1]::regtype in ('sys.varchar', 'sys.nvarchar', 'sys.varbinary', 'sys.nchar','sys.binary','sys.bpchar') THEN '1'
+                                            ELSE typmod::text
+                                        END
+                                    WHEN p2.prokind = 'f' AND p2.proallargtypes IS NULL AND p2.prorettype IS NOT NULL THEN
```

Comment:
Can you explain the usecase of this? Is it possible for PLTSQL function having proargtypes and prorettype both not null?

Reply:
Yes, in case of functions that have input arguments(let's say n arguments) and a return value, we store input argument types in proargtypes and return type in prorettype.
In this case, the size of the typmod array will be n+1 (input + return types). Hence, the first n values, we are picking up from proargtypes and the last one from prorettype